widget/embedded/src/utils/wallets.ts

```diff
@@ -568,3 +479,44 @@ export function getAddress({
       connectedWallet.chain === chain
   )?.address;
 }
+
+export function createTokenHash(token: Asset): TokenHash {
+  return `${token.symbol}-${token.blockchain}-${token.address ?? ''}`;
+}
+
+export function makeTokensBalance(connectedWallets: ConnectedWallet[]) {
+  return connectedWallets
+    .flatMap((wallet) => wallet.balances)
+    .reduce((balances: TokensBalance, balance) => {
+      const currentBalance = {
+        amount: balance?.amount ?? '',
+        decimals: balance?.decimal ?? 0,
+        usdValue: balance?.usdPrice
+          ? new BigNumber(balance?.usdPrice ?? ZERO)
+              .multipliedBy(balance?.amount)
+              .toString()
+          : '',
+      };
+
+      const tokenHash = balance
+        ? createTokenHash({
+            symbol: balance.symbol,
+            blockchain: balance.chain,
+            address: balance.address,
+          })
+        : null;
+
+      const prevBalance = tokenHash ? balances[tokenHash] : null;
```

Comment:
It is possible to add a method to balance store for getting balance of a single token? so we could move this hash generation there instead of where we want to get a token balance in different part of codes.

Reply:
Maybe you could just use `getBalanceFor` here!